config/middlewares/logging_middleware.py

```
@@ -0,0 +1,43 @@
+
+        return response
+
+    def do_request_log(self, request):
+        logger.info(
+            "REQUEST method=%s path=%s ip=%s",
+            request.method,
+            request.path,
+            self._get_client_ip(request),
+        )
+
+    def _get_client_ip(self, request):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `_get_client_ip`

(ANN202)